### PR TITLE
eslint: ban `Record`

### PR DESCRIPTION
**Pull request by Trim21**
mentioned in https://github.com/quisquous/cactbot/pull/3014#discussion_r647110732

### CODE REVIEW

File: resources/party.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default class PartyTracker {
   allianceIds_: string[] = [];
   nameToRole_: { [name: string]: Role } = {};
   idToName_: { [id: string]: string } = {};
-  roleToPartyNames_: Record<Role, string[]> = emptyRoleToPartyNames();
```
Comment (quisquous) [inline resources/party.ts:24]: I think `Record` is fine for cases where the key value is a union type.  It's not that I want to remove all uses of `Record`.  It's more that I think `Record<string, etc>` and `Record<number, etc>` are clearer as index types?

**Comment by Trim21** [comment]
I think there isn't such a rule...

**Comment by quisquous** [comment]
Sorry.  I'm happy to revisit if other people feel differently about using `Record`.  Maybe everybody else disagrees with me thinking it's weird?

**Comment by Trim21** [comment]
looks like this rule could support Record<string, etc

**Pull request closed without merging**